rgd/geodata/serializers.py

```diff
@@ -27,8 +29,55 @@ class Meta:
 class ConvertedImageFileSerializer(serializers.ModelSerializer):
     class Meta:
         model = models.ConvertedImageFile
-        fields = ['source_image']
+        fields = ['source_image', 'pk', 'status', 'failure_reason']
         read_only_fields = ['pk', 'status', 'failure_reason']
 
 
+class ArbitraryFileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.ArbitraryFile
+        fields = [
+            'pk',
+            'file',
+            'validate_checksum',
+            'name',
+            'checksum',
+            'last_validation',
+            'modified',
+            'created',
+        ]
+        read_only_fields = ['pk', 'checksum', 'last_validation', 'modified', 'created']
+
+
+class SubsampledImageSerializer(serializers.ModelSerializer):
+
+    data = serializers.HyperlinkedRelatedField(
+        many=False, read_only=True, view_name='arbitrary-file-data'
+    )
+
+    def to_representation(self, value):
+        ret = super().to_representation(value)
+        # TODO: fix domain
+        ret['status'] = reverse('subsampled-status', args=[value.id])
```

Comment:
I'm working on getting sample data loaded. I think this will work though:

```suggestion
        realtive_status_uri = reverse('subsampled-status', args=[value.id])
        if 'request' in self.context:
            request = self.context['request']
            ret['status'] = request.build_absolute_uri(realtive_status_uri)
        else:
            ret['status'] = realtive_status_uri
```

Reply:
Just tested and this works.

Reply:
See https://github.com/ResonantGeoData/ResonantGeoData/pull/217/commits/cf33017ba9bb70734d88e690d8f6b0bbff9441e7.

Reply:
Awesome, thanks for figuring this out!